src/tests/pmfs_demo.c

```diff
@@ -0,0 +1,1528 @@
+					     argv2, S_IFREG, O_CREAT | O_RDWR,
+					     atoi(argv3), argv4, &g_present_pool->p_cmd_obj);
+			if (rc != 0) {
+				D_PRINT("pmfs open file failed\r\n");
+				continue;
+			}
+			D_PRINT("create file obj=%p\r\n", g_present_pool->p_cmd_obj);
+
+			continue;
+		} else if (check_cmd(argv1, "r ") || check_cmd(argv1, "readsync ")) {
+			then = dts_time_now();
+
+			rc = pmfs_read_start(g_present_pool->p_cmd_pmfs, g_present_pool->p_cmd_obj,
+					     &g_user_sgl, atoll(argv2), &g_read_size);
+			if (rc != 0) {
+				D_PRINT("pmfs read file failed\r\n");
+				continue;
+			}
+			now = dts_time_now();
+			D_PRINT("read file size=%ld\r\n", g_read_size);
+			D_PRINT("last time for whole cmd= %-10.6fs\r\n", now-then);
+			D_PRINT("read speed= %-10.3f  MiB/sec\r\n",
+				g_read_size/(now-then)/(1024*1024));
+			D_PRINT("pmfs read file done\r\n");
+			continue;
+		} else if (check_cmd(argv1, "w ") || check_cmd(argv1, "writesync ")) {
+			then = dts_time_now();
+			g_write_size = atoll(argv3);
+
+			rc = pmfs_write_start(g_present_pool->p_cmd_pmfs, g_present_pool->p_cmd_obj,
+					      &g_user_sgl, atoll(argv2), &g_write_size);
+			if (rc != 0) {
+				D_PRINT("pmfs write file failed\r\n");
+				continue;
+			}
+			now = dts_time_now();
+			D_PRINT("write file obj=%p\r\n", &g_user_sgl);
+			D_PRINT("last time for whole cmd= %-10.6fs\r\n", now-then);
+			D_PRINT("write speed= %-10.3f  MiB/sec\r\n",
+				(g_write_size - atoll(argv2))/(now-then)/(1024*1024));
+			D_PRINT("pmfs write file done\r\n");
+			continue;
+		} else if (check_cmd(argv1, "p ") || check_cmd(argv1, "punch ")) {
+			then = dts_time_now();
+			g_punch_size = atoll(argv3);
+			if (atoll(argv2) > g_write_size) {
+				D_PRINT("pmfs punch offset is over file size\r\n");
+				continue;
+			}
+			if (g_punch_size + atoll(argv2) > g_write_size) {
+				D_PRINT("pmfs punch size over file size");
+				D_PRINT("punch size equals filesize-offset\r\n");
+				g_punch_size = g_write_size - atoll(argv2);
+			}
+			rc = pmfs_punch_start(g_present_pool->p_cmd_pmfs,
+					      g_present_pool->p_cmd_obj,
+					      atoll(argv2), g_punch_size);
+			if (rc != 0) {
+				D_PRINT("pmfs punch file failed\r\n");
+				continue;
+			}
+			now = dts_time_now();
+			D_PRINT("pmfs punch file done\r\n");
+			D_PRINT("last time for whole cmd= %-10.6fs\r\n", now-then);
+			D_PRINT("punch speed= %-10.3f  MiB/sec\r\n",
+					g_punch_size/(now-then)/(1024*1024));
+			continue;
+		} else if (check_cmd(argv1, "ls ")) {
+			mode_t mode;
+			struct pmfs_obj *tmp_obj = g_present_pool->p_cmd_obj;
+
+			rc = pmfs_lookup_start(g_present_pool->p_cmd_pmfs, argv2,
+					       O_RDONLY, &tmp_obj, &mode, NULL);
+			if (rc != 0  || !tmp_obj) {
+				D_PRINT("no such file or dir %s\r\n", argv2);
+				continue;
+			}
+
+			uint32_t nr, i;
+
+			rc = pmfs_listdir(g_present_pool->p_cmd_pmfs, tmp_obj, &nr);
+			D_PRINT("start to readdir\r\n");
+			rc = pmfs_readdir_start(g_present_pool->p_cmd_pmfs, tmp_obj,
+					 	&nr, tmp_dir);
```

Comment:
(style)  code indent should use tabs where possible
(style)  please, no space before tabs